src/panels/lovelace/card-features/hui-button-card-feature.ts

```diff
@@ -50,15 +53,29 @@ class HuiButtonCardFeature extends LitElement implements LovelaceCardFeature {
 
     if (domain === "script") {
       const entityId = this._stateObj.entity_id;
-      if (hasScriptFields(this.hass!, entityId)) {
-        showMoreInfoDialog(this, { entityId: entityId });
+      if (
+        hasRequiredScriptFields(this.hass!, entityId) &&
+        (!this._config?.data ||
+          !requiredScriptFieldsFilled(this.hass!, entityId, this._config?.data))
```

Comment:
`this._config?.data` is already checked in `requiredScriptFieldsFilled`

```suggestion
        !requiredScriptFieldsFilled(this.hass!, entityId, this._config?.data)
```